src/ui/public/timepicker/timepicker.js

```diff
@@ -20,14 +20,14 @@ module.directive('kbnTimepicker', function (quickRanges, timeUnits, refreshInter
   return {
     restrict: 'E',
     scope: {
-      from: '=',
-      to: '=',
+      time: '=',
       mode: '=',
       interval: '=',
       activeTab: '='
     },
     template: html,
-    controller: function ($scope) {
+    require: '^kbnTopNav',
```

Comment:
I would advise to avoid introducing such a dependency on the hierarchy the timepicker is placed in, because it limits reuse and couples the components more tightly. Why not add a `hide: '&'` attribute that can be used by the parent component to pass a function as in `<kbn-timepicker ... hide="kbnTopNav.close('filter')">`. This way the `<kbn-timepicker>` directive can be used in contexts where there is no `kbnTopNav`.

Even better would be to move from implicitly modifying the values to an explicit callback (e.g. `change(newFrom, newTo)` with `change` having been passed in by the parent) and let the parent control the visibility, but that might be out of scope for this PR.